stumpy/scrump.py

```diff
@@ -103,67 +109,113 @@ def _compute_PI(
     for i in indices[start:stop]:
         Q = T_A[i : i + m]
         QT[:] = core._sliding_dot_product(Q, T_B)
-        # Update P[i] relative to all T[j : j + m]
         squared_distance_profile[:] = core._mass(Q, T_B, QT, μ_Q[i], σ_Q[i], M_T, Σ_T)
         squared_distance_profile[:] = np.square(squared_distance_profile)
         if excl_zone is not None:
             zone_start = max(0, i - excl_zone)
             zone_stop = min(l, i + excl_zone)
             squared_distance_profile[zone_start : zone_stop + 1] = np.inf
 
-            # only for self-join
-            mask = squared_distance_profile < P_squared[thread_idx]
-            P_squared[thread_idx][mask] = squared_distance_profile[mask]
-            I[thread_idx][mask] = i
+        if excl_zone is not None:  # self-join
```

Comment:
I created this redundant `if-condition` to improve readability. What do you think? Also, how about the comments after that? Does it clarify what's going on?

Reply:
No, we should **not** include this redundant `if` condition. All we need is a newline inbetween the sections as it is, provide the comment after the newline, and then the code

I now understand what you are trying to accomplish in the code! I think it could be shortened and cleared up in notation:

```
# Note that the squared distance, `squared_distance_profile[j]`,
# between subsequences `S_i = T[i : i + m]` and `S_j = T[j : j + m]`
# can be used to update the top-k for BOTH subsequence `i` and 
# subsequence `j`. We update the latter here. 
```

Reply:
> I think it could be shortened and cleared up in notation:

Yeah...it is cleaner indeed!